cmake/PythonLinkHelper.cmake

```diff
@@ -0,0 +1,25 @@
+# =============================================================================
+# Support for building NMODL without python library
+# =============================================================================
+# ~~~
+# When NMODL is built under environment  (e.g. manylinux) without python-dev
+# package i.e. libpython.so, we have to build extension and binaries by
+# ignoring undefned symbols. The python library will be loaded later on target
+# system.
+# ~~~
+
+# use same variable name as NEURON as it won't be user option
+set(LINK_AGAINST_PYTHON
+    TRUE
+    CACHE BOOL "Disable linking to python library")
+
+# Flags for ignoring undefined symbols for wheel
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(UNDEFINED_SYMBOLS_IGNORE_FLAG "-Wl,-undefined dynamic_lookup")
+else()
+  set(UNDEFINED_SYMBOLS_IGNORE_FLAG "-Wl,--unresolved-symbols=ignore-all")
+endif()
+
+if(NOT LINK_AGAINST_PYTHON)
```

Comment:
https://pybind11.readthedocs.io/en/stable/compiling.html

> Since pybind11 is a metatemplate library, it is crucial that certain compiler flags are provided to ensure high quality code generation. In contrast to the pybind11_add_module() command, the CMake interface library only provides the minimal set of parameters to ensure that the code using pybind11 compiles, but it does not pass these extra compiler flags (i.e. this is up to you).
> 
> These include Link Time Optimization (-flto on GCC/Clang/ICPC, /GL and /LTCG on Visual Studio) and .OBJ files with many sections on Visual Studio (/bigobj). The FAQ contains an explanation on why these are needed.

I think we should make sure `-flto` is added at least when compiling `.cpp`s that include pybind11

Reply:
note what we have documented in [src/pybind/CMakeLists.txt](https://github.com/BlueBrain/nmodl/blob/987dbc83b459b74a251db284a214025ca9465655/src/pybind/CMakeLists.txt#L34)

```
# Note that LTO causes link time errors with GCC 8. To avoid this, we disable LTO for pybind using
# NO_EXTRAS. See #266.
pybind11_add_module(
```

Reply:
my bad - have you checked that this issue is still persisting?

Reply:
We didn't change pybind11 so I believe the same issue should exist. But I can check that separately.